examples/dao/test/dao-app-flow.js

```diff
@@ -0,0 +1,457 @@
+
+    const daoFundLsigProg = getProgram('dao-fund-lsig.py', scInitParam);
+    daoFundLsig = runtime.getLogicSig(daoFundLsigProg, []);
+    daoFundLsigAcc = runtime.getAccount(daoFundLsig.address());
+
+    const proposalLsigProg = getProgram('proposal-lsig.py',
+      { ARG_OWNER: proposer.address, ARG_DAO_APP_ID: applicationID });
+    proposalLsig = runtime.getLogicSig(proposalLsigProg, []);
+    proposalLsigAcc = runtime.getAccount(proposalLsig.address());
+
+    // fund lsig's
+    for (const lsig of [depositLsig, daoFundLsig, proposalLsig]) {
+      runtime.fundLsig(master.account, lsig.address(), minBalance + 10000);
+    }
+    syncAccounts();
+
+    // verify global state
+    assert.isDefined(applicationID);
+    assert.deepEqual(getGlobal('deposit'), BigInt(deposit));
+    assert.deepEqual(getGlobal('min_support'), BigInt(minSupport));
+    assert.deepEqual(getGlobal('min_duration'), BigInt(minDuration));
+    assert.deepEqual(getGlobal('max_duration'), BigInt(maxDuration));
+    assert.deepEqual(getGlobal('url'), parsing.stringToBytes(url));
+
+    // add deposit_lsig address to DAO
+    const addAccountsTx = {
+      type: types.TransactionType.CallApp,
+      sign: types.SignType.SecretKey,
+      fromAccount: creator.account,
+      appID: applicationID,
+      payFlags: {},
+      appArgs: [
+        'str:add_deposit_accounts',
+        `addr:${depositLsig.address()}`
+      ]
+    };
+    runtime.executeTx(addAccountsTx);
+
+    // verify deposit_lsig address was added
+    assert.deepEqual(getGlobal('deposit_lsig'), convert.addressToPk(depositLsig.address()));
+
+    // optIn to ASA(Gov Token) by accounts
+    for (const acc of [proposer, voterA, voterB, depositLsigAcc, daoFundLsigAcc]) {
+      runtime.optIntoASA(govTokenID, acc.address, {});
+    }
+    syncAccounts();
+
+    // verify optIn
+    for (const acc of [proposer, voterA, voterB, depositLsigAcc, daoFundLsigAcc]) {
+      assert.isDefined(acc.getAssetHolding(govTokenID));
+    }
+
+    // GOV Token distribution (used as initial fund)
+    const distributeGovTokenParams = {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: creator.account,
+      amount: 100,
+      assetID: govTokenID,
+      payFlags: { totalFee: 1000 }
+    };
+
+    runtime.executeTx([
+      { ...distributeGovTokenParams, toAccountAddr: proposer.address },
+      { ...distributeGovTokenParams, toAccountAddr: voterA.address },
+      { ...distributeGovTokenParams, toAccountAddr: voterB.address }
+    ]);
+    syncAccounts();
+
+    assert.equal(proposer.getAssetHolding(govTokenID).amount, 100);
+    assert.equal(voterA.getAssetHolding(govTokenID).amount, 100);
+    assert.equal(voterB.getAssetHolding(govTokenID).amount, 100);
+
+    /* --------------------  Add proposal  -------------------- */
+
+    // optIn to DAO by proposalLsig
+    runtime.optInToApp(proposalLsig.address(), applicationID, {}, {});
```

Comment:
In `proposal.lsig` a check has been added to allow optIn only from owner, but still this call is passing.
could you please check ?

Reply:
do we need opt-in for asset in proposal lsig ?

Reply:
> do we need opt-in for asset in proposal lsig ?

yes, during clear_proposal we take gov_tokens back in `proposalLsig`.

Added optInLock for AppOptIn as well.